### DISCUSSION TITLE
HW 1 Part 2 questions 2 and 3 the same?

### DISCUSSION THREAD

**Discussion by ecieszynski** [cooperrc/computational-mechanics#28] [Q&A]
I believe the answers for 2 and 3 should be the same, the only thing that changes is the methodology right? So we should be finding a 10x10 2D array for both right? This is what I found for 3 for reference, just want to make sure I understand number 2 because the problem statement isn't clear.

![image](https://user-images.githubusercontent.com/106542680/171762650-2d0e8c17-82c9-442a-8060-12d6a6a77b25.png)

**Comment by FishMacaron** [comment]
I think so! I got the same mean and standard deviation. I believe the purpose of this problem is just so we can compare how much time both processes take with how long it takes to write the code and for python to process it.

**Reply by ecieszynski** [reply]
perfect, thank you. Any tips for starting number 2? I'm kinda lost on how to do this one

**Reply by FishMacaron** [reply]
I heavily referenced the "Element-wise sum of a Python list" section for this problem because of its for loop- I'll try and drop some hints for how I did it:
1) my code has a similar format to the example that I mentioned above but I ended up putting a for loop within a for loop 
2) you'll end up with the same array as the one you got in the original post, I'd print the result each time to see if your A_99 matches
3) you are still going to append in the for loop but make sure it's a product

Maybe those are weird hints, if it's still confusing or you'd like to show your progress I'd be happy to continue helping out!

**Reply by UniquelyAbbi** [reply]
I also agree on being lost. Im on number 2 of part two but im not sure im doing it right. I got to this part using the book and some advice but ... I need a bit more explanation. Sorry for the bother
![image](https://user-images.githubusercontent.com/106557922/171770861-08f53395-549c-4080-b238-52d62739b08e.png)

**Reply by FishMacaron** [reply]
@UniquelyAbbi it's no problem! I did the exact same process as you did when I started this question.
For this problem, you don't need to use *import random* at all. The end product of A_99 is an array with all the products of 0-9. In the original post with the huge array it starts with a bunch of 0s because it is 0\*0, 0\*1, ... 0\*9 and then moves onto 1\*0, 1\*1, ... 1\*9, etc. You're trying to create this in a for loop so you'll range will be 10 (10 because it starts at 0! You need values 0-9). No need for random numbers at all. Everything above the A_99=[] can be deleted. 

As for your for loop, you have the right idea but you'll need to change the range (reference when I said your range should be 10!). But one for loop won't be enough to create the array, hopefully hint 1 from above helps with that. Also see hint 3 from my previous comment for your appending!!

Let me know if you need clarification with anything I mentioned or if you'd like more feedback.

**Reply by UniquelyAbbi** [reply]
@FishMacaron thank you ill keep that in mind

**Reply by cooperrc** [reply]
I've seen a couple people use `random` here, but there's no need to randomize the values in the lists. 

You're creating a multiplication table $[0...9]\times[0...9]$

there are many different approaches, but for each of them you need two `for`-loops

```python
for i in range(10):
    for j in range(10):
        i*j
```
You can use [list comprehension](https://realpython.com/list-comprehension-python/) or the [`append`](https://realpython.com/python-append/) approach. 

List comprehension is the more "Pythonic" way of doing things.
